### ISSUE TITLE
Make ores and their places dynamic per match

### ISSUE THREAD

**Issue by farooqkz** [MT-CTF/capturetheflag#1165]
Currently in every single match, location of ores is the same for each map and players simply know where to look for them. I think we can make their location dynamic, that is set per match at the beginning of it. Thus, players who are already familiar with the map, won't have any advantage over other players. Also this adds more adventure to the game.

Regarding implementation, I suggest iterating over all destructible stone blocks on the map and using some simple heuristic and a probability, decide to convert the stone to an ore, or not.

**Comment by src4026** [comment]
Would it allow for regions too, similar to how chest zones work?

**Comment by farooqkz** [comment]
> Would it allow for regions too, similar to how chest zones work?

Yeah the implementation would allow that. But we'll need a new parameter in `map.conf`. And I want LW's comment on this as well.

**Comment by src4026** [comment]
> > Would it allow for regions too, similar to how chest zones work?
> 
> Yeah the implementation would allow that. But we'll need a new parameter in `map.conf`. And I want LW's comment on this as well.

Indeed. I was just asking :)

**Comment by LoneWolfHT** [comment]
You could get a list of all ore nodes on the map and randomly swap them around. I think ore clusters would have to be sacrificed to do this simply anyway

**Comment by farooqkz** [comment]
> You could get a list of all ore nodes on the map and randomly swap them around. I think ore clusters would have to be sacrificed to do this simply anyway

We can do it NOT the simplest way :)

**Comment by LoneWolfHT** [comment]
You can still start off with what I suggest, then sort all the node positions into groups based on how close they are to each other. Would need to double-check how it affects The Wall map
